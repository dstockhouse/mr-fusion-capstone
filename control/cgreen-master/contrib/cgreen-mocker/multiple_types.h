typedef struct Struct1 {
    int someValue;
} Struct1;

typedef Struct1* Struct1Ptr;

Struct1 return_struct1_by_value(int i);
Struct1Ptr return_pointer_to_struct1(char string[]);
Struct1* direct_return_pointer_to_struct1(void);

typedef struct Struct2 {
    int someValue;
} Struct2;

typedef Struct2* Struct2Ptr;

Struct2 return_struct2_by_value(int i);
Struct2Ptr return_pointer_to_struct2(char string[]);
Struct2* direct_return_pointer_to_struct2(void);

extern void return_nothing(void);
extern int return_int(int ints[]);
