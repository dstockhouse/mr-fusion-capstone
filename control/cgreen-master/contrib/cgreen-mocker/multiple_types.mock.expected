/* -*- c -*-*/
#include "multiple_types.h"
#include <cgreen/mocks.h>

Struct1 return_struct1_by_value(int i) { 
  return *(Struct1 *) mock(i);
}

Struct1Ptr return_pointer_to_struct1(char string[]) { 
  return (Struct1Ptr) mock(string);
}

Struct1 *direct_return_pointer_to_struct1(void) { 
  return (Struct1 *) mock();
}

Struct2 return_struct2_by_value(int i) { 
  return *(Struct2 *) mock(i);
}

Struct2Ptr return_pointer_to_struct2(char string[]) { 
  return (Struct2Ptr) mock(string);
}

Struct2 *direct_return_pointer_to_struct2(void) { 
  return (Struct2 *) mock();
}

void return_nothing(void) { 
  mock();
}

int return_int(int ints[]) { 
  return (int) mock(ints);
}

